tests/components/thethingsnetwork/conftest.py

```diff
@@ -0,0 +1,100 @@
+"""Define fixtures for the The Things Network tests."""
+
+from unittest.mock import AsyncMock, patch
+
+import pytest
+from ttn_client import TTNSensorValue
+
+from homeassistant.components.thethingsnetwork.const import (
+    CONF_API_KEY,
+    CONF_APP_ID,
+    CONF_HOSTNAME,
+    DOMAIN,
+    TTN_API_HOSTNAME,
+)
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry
+
+HOSTNAME = "example.com"
+APP_ID = "my_app"
+API_KEY = "my_api_key"
+
+DEVICE_ID = "my_device"
+DEVICE_ID_2 = "my_device_2"
+DEVICE_FIELD = "a_field"
+DEVICE_FIELD_2 = "a_field_2"
+DEVICE_FIELD_VALUE = 42
+
+DATA = {
+    DEVICE_ID: {
+        DEVICE_FIELD: TTNSensorValue(
+            {
+                "end_device_ids": {"device_id": DEVICE_ID},
+                "received_at": "2024-03-11T08:49:11.153738893Z",
+            },
+            DEVICE_FIELD,
+            DEVICE_FIELD_VALUE,
+        )
+    }
+}
+
+DATA_UPDATE = {
+    DEVICE_ID: {
+        DEVICE_FIELD: TTNSensorValue(
+            {
+                "end_device_ids": {"device_id": DEVICE_ID},
+                "received_at": "2024-03-12T08:49:11.153738893Z",
+            },
+            DEVICE_FIELD,
+            DEVICE_FIELD_VALUE,
+        )
+    },
+    DEVICE_ID_2: {
+        DEVICE_FIELD_2: TTNSensorValue(
+            {
+                "end_device_ids": {"device_id": DEVICE_ID_2},
+                "received_at": "2024-03-12T08:49:11.153738893Z",
+            },
+            DEVICE_FIELD_2,
+            DEVICE_FIELD_VALUE,
+        )
+    },
+}
+
+CONFIG_ENTRY = MockConfigEntry(
+    domain=DOMAIN,
+    unique_id=APP_ID,
+    title=APP_ID,
+    data={
+        CONF_APP_ID: APP_ID,
+        CONF_HOSTNAME: TTN_API_HOSTNAME,
+        CONF_API_KEY: API_KEY,
+    },
+)
+
+
+@pytest.fixture
+async def init_integration(hass: HomeAssistant):
```

Comment:
Just make it a function, not a fixture. A function gives you just that extra bit of flexilibity of for example patching the library before you setup the integration. I have tried this way before a couple of times, and it always bit me back, so I highly recommend just creating this a function